processor/resourcedetectionprocessor/internal/azure/aks/aks.go

```diff
@@ -0,0 +1,68 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aks
+
+import (
+	"context"
+	"os"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/azure"
+)
+
+const TypeStr = "aks"
+
+type Detector struct {
+	provider azure.Provider
+}
+
+// NewDetector creates a new AKS detector
+func NewDetector(component.ProcessorCreateParams, internal.DetectorConfig) (internal.Detector, error) {
+	return &Detector{provider: azure.NewProvider()}, nil
+}
+
+func (d *Detector) Detect(ctx context.Context) (pdata.Resource, error) {
+	res := pdata.NewResource()
+
+	if !onK8s() {
+		return res, nil
+	}
+
+	// If we can't get a response from the metadata endpoint, we're not running in Azure
+	if !azureMetadataAvailable(ctx, d.provider) {
+		return res, nil
+	}
+
+	attrs := res.Attributes()
+	attrs.InsertString(conventions.AttributeCloudProvider, conventions.AttributeCloudProviderAzure)
+	attrs.InsertString(conventions.AttributeCloudPlatform, conventions.AttributeCloudPlatformAzureAKS)
+
+	return res, nil
+}
+
+func onK8s() bool {
+	// Environment variable that is set when running on Kubernetes
+	const kubernetesServiceHostEnvVar = "KUBERNETES_SERVICE_HOST"
```

Comment:
I'd put this up at top of the file like eks since having it here doesn't seem much different than just calling Getenv("KUBERNETES_SERVICE_HOST"). (personally I prefer once-used strings to be inline but others disagree)

Reply:
Sure, done.